src/Sentry.AspNetCore/SentryTunnelMiddleware.cs

```diff
@@ -98,10 +98,14 @@ public async Task InvokeAsync(HttpContext context, RequestDelegate next)
                     Method = HttpMethod.Post,
                     Content = new StreamContent(memoryStream),
                 };
+                var existingForwardedFor = context.Request.Headers["X-Forwarded-For"];
                 var clientIp = context.Connection?.RemoteIpAddress?.ToString();
                 if (clientIp != null)
                 {
-                    sentryRequest.Headers.Add("X-Forwarded-For", context.Connection?.RemoteIpAddress?.ToString());
+                    var forwardedFor = string.IsNullOrEmpty(existingForwardedFor)
+                        ? clientIp
+                        : $"{existingForwardedFor}, {clientIp}";
```

Comment:
question: Would `Add` to the `"X-Forwarded-For"` header work as well?
Does `X-Forwarded-For` need to be a single comma-separated string, or could this be multiple values?

Could we maybe avoid the extra string-allocation, and do something like
```CSharp
if (existingForwardedFor.Count == 1)
{
    sentryRequest.Headers.Add("X-Forwarded-For", existingForwardedFor.ToString());
}
else if (existingForwardedFor.Count > 1)
{
    sentryRequest.Headers.Add("X-Forwarded-For", existingForwardedFor.ToArray());
}
sentryRequest.Headers.Add("X-Forwarded-For", clientIp);
```

Reply:
There's no Add method on the StringValues type (it's a struct)... but we can maybe leverage the `Count` to create some tiny optimisations.